src/bootstrap/bootstrap.py

```diff
@@ -638,8 +638,10 @@ def fix_bin_or_dylib(self, fname, rpath_libz=False):
     # Return the stage1 compiler to download, if any.
     def maybe_download_rustc(self):
         # If `download-rustc` is not set, default to rebuilding.
-        if self.get_toml("download-rustc", section="rust") != "true":
+        download_rustc = self.get_toml("download-rustc", section="rust")
+        if download_rustc is None or download_rustc == "false":
             return None
+        assert download_rustc == "true" or download_rustc == "if-unchanged", download_rustc
```

Comment:
How nice is the error message here in terms of noting download_rustc at least?

Reply:
It looks like this:
```
Traceback (most recent call last):
  File "./x.py", line 27, in <module>
    bootstrap.main()
  File "/home/joshua/rustc2/src/bootstrap/bootstrap.py", line 1181, in main
    bootstrap(help_triggered)
  File "/home/joshua/rustc2/src/bootstrap/bootstrap.py", line 1142, in bootstrap
    build.rustc_commit = build.maybe_download_rustc()
  File "/home/joshua/rustc2/src/bootstrap/bootstrap.py", line 644, in maybe_download_rustc
    assert download_rustc == "true" or download_rustc == "if-unchanged", download_rustc
AssertionError: if-available
```
I think that's pretty good, it shows you both allowed options as well as the actual value

Reply:
Agreed. r=me